src/YMenuWidget.cc

```diff
@@ -0,0 +1,293 @@
+            {
+                conflicts.push_back(item);
+                yuiMilestone() << "No or invalid shortcut found: \""
+                               << item->label() << "\""
+                               << endl;
+            }
+            else if ( used[ (unsigned) shortcut ] )
+            {
+                conflicts.push_back(item);
+
+                yuiWarning() << "Conflicting shortcut found: \""
+                             << item->label() << "\""
+                             << endl;
+            }
+            else
+            {
+                used[ (unsigned) shortcut ] = true;
+            }
+	}
+        else
+        {
+            yuiWarning() << "Non-menu item used in call: \""
+                         << (*it)->label() << "\""
+                         << endl;
+        }
+    }
+
+    // Cannot use YShortcut directly as an YItem is not a YWidget
+
+    for ( YMenuItem *i: conflicts )
+    {
+        string clean = YShortcut::cleanShortcutString(i->label());
+        char new_c = 0;
+
+        size_t index = 0;
+        for (; index < clean.size(); ++index)
+        {
+            char ch = YShortcut::normalized( clean[ index ] );
+
+            // ch is set to 0 by normalized() if not valid
+            if ( ch != 0 && ! used[ (unsigned) ch ] )
+            {
+                new_c = ch;
+                used[ (unsigned) ch ] = true;
+                break;
+            }
+        }
+
+        if ( new_c != 0 )
+        {
+            clean.insert( index, 1, YShortcut::shortcutMarker() );
+            yuiMilestone() << "New label used: " << clean << endl;
+        }
+
+        i->setLabel( clean );
+    }
+}
+
+// FIXME End
+
+
+void
+YMenuWidget::resolveShortcutConflicts()
+{
+    resolveShortcutConflicts( itemsBegin(), itemsEnd() );
+}
+
+
+YMenuItem *
+YMenuWidget::findItem( std::vector<std::string> & path ) const
+{
+    return findItem( path.begin(), path.end(),
+                     itemsBegin(), itemsEnd() );
+}
+
+
+YMenuItem *
+YMenuWidget::findItem( std::vector<std::string>::iterator path_begin,
+                       std::vector<std::string>::iterator path_end,
+                       YItemConstIterator                 begin,
+                       YItemConstIterator                 end ) const
+{
+    for ( YItemConstIterator it = begin; it != end; ++it )
+    {
+        YMenuItem * item = dynamic_cast<YMenuItem *>(*it);
+        // Test that dynamic_cast didn't fail
+
+        if ( !item )
+            return 0;
+
+        if ( item->label() == *path_begin )
+        {
+            if ( std::next(path_begin) == path_end )
+            {
+                // Only return items which can trigger an action.
+                // Intermediate items only open a submenu, so continue looking.
+                if( item->hasChildren() )
+                    continue;
+
+                return item;
+            }
+
+            // Look in child nodes and return if found one
```

Comment:
And here.

Reply:
Not sure what the original author meant to say with this, but it only creates confusion instead of helping to understand. Cut off the nonsensical part.